src/Altinn.Apps/AppFrontend/react/altinn-app-frontend/src/utils/validation.ts

```diff
@@ -384,6 +384,20 @@ function attachmentsValid(attachments: any, component: any): boolean {
   );
 }
 
+export function getRuleBasedDate(date: string, defaultDate?: string): moment.Moment | string {
+  if (!date) {
+    return defaultDate ?? null;
+  }
+
+  if (date === DatepickerRestrictionFlags.Today) {
```

Comment:
Consider using something other than `date` for a variable name that can be either `'today'` or a valid date that can be parsed by `moment`.
The behavior of this method can become very unstable, and it invites for extending with more `DatepickerRestrictionFlags`.
Consider simplifying the output you can expect from the function in general.
Right now it can return a `moment.Moment` date, `string` which is already known outside the function or `null`. Null should be defined behavior, however it's never used for something definite.